src/faucet/Faucet.sol

```diff
@@ -0,0 +1,90 @@
+/*HOLOGRAPH_LICENSE_HEADER*/
+
+/*SOLIDITY_COMPILER_VERSION*/
+
+import "../abstract/Initializable.sol";
+import "../interface/ERC20Holograph.sol";
+import "../interface/IInitializable.sol";
+
+contract Faucet is Initializable {
+  address public owner;
+  ERC20Holograph public token;
+
+  uint256 public faucetDripAmount = 100 ether;
+  uint256 public faucetCooldown = 24 hours;
+
+  mapping(address => uint256) lastAccessTime;
+
+  /**
+  * @notice Constructor is empty and not utilised.
+  * @dev To make exact CREATE2 deployment possible, constructor is left empty. We utilize the "init" function instead.
+  */
+  constructor() {}
+
+  /**
+  * @notice Initializes the token.
+  * @dev Special function to allow a one time initialisation on deployment.
+  */
+  function init(bytes memory data) external override returns (bytes4) {
+    require(!_isInitialized(), "Faucet already initialized");
+    (address contractOwner_, address tokenInstance_) = abi.decode(data, (address, address));
+    token = ERC20Holograph(tokenInstance_);
+    owner = contractOwner_;
+    _setInitialized();
+    return IInitializable.init.selector;
+  }
+
+  /// @notice Get tokens from faucet's own balance. Rate limited.
+  function requestTokens() external {
+    require(isAllowedToWithdraw(msg.sender), 'Come back later');
+    require(token.balanceOf(address(this)) >= faucetDripAmount, "Faucet is empty");
```

Comment:
does this mean the faucet can only have 100 tokens at a time?

Reply:
It means that if faucet's balance is not larger or equal the drip amount, then it will revert.

Reply:
Okay cool makes sense